web/src/context/AtlasProvider.tsx

```diff
@@ -94,16 +96,30 @@ const AtlasProvider: React.FC<{ children?: React.ReactNode }> = ({ children }) =
   }, [authToken, address]);
 
   useEffect(() => {
-    // initial verfiy check
-    setIsVerified(verifySession());
+    let timeoutId: ReturnType<typeof setTimeout>;
+
+    const verifyAndSchedule = () => {
+      // initial verfiy check
+      const isValid = verifySession();
+      setIsVerified(isValid);
+
+      if (isValid && authToken) {
+        try {
+          const payload = decodeJwt(authToken);
+          const expiresIn = (payload.exp as number) * 1000 - Date.now();
```

Comment:
Yep, I thought about that, but the `verifySession()` just verifies the current auth token, it does not initiate re-verification, I could just do `setIsVerified` to false too

Even so, it would be uncommon as you said and in case it does occur , we have session verification if a call returns `UNAUTHENTICATED` code.